### ISSUE TITLE
Bug - Dropdown - Cannot select dropdown options or read options with tabs/arrows

### ISSUE THREAD

**Issue by agagancarczyk** [patternfly/patternfly#5766]
**Describe the problem**
A clear and concise description of the problem. Which components are affected?
The user is unable to select options from the dropdown and options are not read when using tabs/arrows.

**How do you reproduce the problem?**
This problem can be reproduced in the Keycloak Admin UI with the following steps:

- Click on Identity Providers from the left-side menu
- Add a new identity provider
- Navigate to the Add provider dropdown and use Chromvox to listen to the options as you use tabs/arrows to navigate between them
- Observe that tab/arrow navigation is not working and the options are not read

**Expected behavior**
The user is able to select options from the dropdown and options are read when using tabs/arrows.

**Is this issue blocking you?**
None. The component is not fully accessible and needs to be fixed.

**Screenshots**
N/A

**What is your environment?**
 - OS: MacBook OS
 - Chrome Browser (the latest)

**What is your product and what release date are you targeting?**
Keyckoak Admin UI, as soon as possible

**Any other information?**

**Comment by jessiehuff** [comment]
Hi @agagancarczyk! I'd love to understand this issue better, is this specific to Chromevox?

**Comment by agagancarczyk** [comment]
> Hi @agagancarczyk! I'd love to understand this issue better, is this specific to Chromevox?

hi @jessiehuff thanks for a follow-up, it is actually possible that this is related to both Chromevox and VoiceOver (as I often test on both). Please check on both, Chromevox and VoiceOver when replicating the steps above.

**Comment by jessiehuff** [comment]
When I test on https://www.patternfly.org/v4/components/dropdown, I don't see the issue present in JAWS, NVDA, VO, or Chromevox. If you can help me get access to the Keycloak Admin UI, perhaps I can help you debug there?

**Comment by agagancarczyk** [comment]
Hi @jessiehuff Thanks, yes let's schedule a quick meeting to discuss this issue further and give you access to the Admin UI.

**Comment by thatblindgeye** [comment]
@agagancarczyk It looks like the issue might be tied to how Keycloak is passing into the `dropdownItems` prop. I've only tested on your PageHeader component, but also tested something on PatternFly's deprecated Dropdown example similar to what Keycloak is doing and noticed the same issue.

I was able to get the dropdown in PageHeader to work correctly by making several changes:

- Rather than passing in an array of the components created inside PageHeader (ManagedAccountDropdownItem, etc), I passed in a function that returned an array of DropdownItems (so instead of having 3-4 individual __DropdownItem components where `const {t} = ...` is destructured in each instance, just destructuring everything needed once in the function, dynamically creating the `key` value if an `isKebab` arg is true.
- The "Realm Info" dropdown item was the only issue after that, since it's not passing the expected `component="a"` and `href="..."` values. So even after doing the above bullet, this item would get skipped in keyboard navigation. I did try updating this DropdownItem to pass in `component="a"` and `href={toDashboard({ realm }).pathname}`, but not entirely sure if that's the intended behavior.

What Keycloak is doing:

![image](https://github.com/patternfly/patternfly/assets/70952936/1491aeed-777c-4801-b02d-de9364b0d5ed)
And passing `dropdownItems={userDropdownItems}` in the Dropdown component.

Part of the helper function I threw together:

![image](https://github.com/patternfly/patternfly/assets/70952936/88d4b8a9-84ed-42ac-8b2d-64bc99ea7320)
And passing `dropdownItems={testDropdownItems(false)}` to the Dropdown component

So a fix might be doable with the v4 implementation of Dropdown, but again this was only testing in the PageHeader component so it'd need testing elsewhere the PF4 Dropdown is being used.

**Comment by mcoker** [comment]
Hey @agagancarczyk! Do you still need help with this issue or is it OK to close?

**Comment by mcoker** [comment]
Closing for now - @agagancarczyk feel free to re-open if you want to re-visit.
